lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -65,14 +65,40 @@ public static function get_theme_data( $deprecated = array(), $settings = array(
 		if ( ! $settings['with_supports'] ) {
 			return static::$theme;
 		}
+		
+		$editor_settings = array(
+			'disableCustomColors'    => get_theme_support( 'disable-custom-colors' ),
+			'disableCustomFontSizes' => get_theme_support( 'disable-custom-font-sizes' ),
+			'disableCustomGradients' => get_theme_support( 'disable-custom-gradients' ),
+			'disableLayoutStyles'    => get_theme_support( 'disable-layout-styles' ),
+			'enableCustomLineHeight' => get_theme_support( 'custom-line-height' ),
+			'enableCustomSpacing'    => get_theme_support( 'custom-spacing' ),
+			'enableCustomUnits'      => get_theme_support( 'custom-units' ),
+		);
+
+		// Theme settings.
+		$color_palette = current( (array) get_theme_support( 'editor-color-palette' ) );
+		if ( false !== $color_palette ) {
+			$editor_settings['colors'] = $color_palette;
+		}
+
+		$font_sizes = current( (array) get_theme_support( 'editor-font-sizes' ) );
+		if ( false !== $font_sizes ) {
+			$editor_settings['fontSizes'] = $font_sizes;
+		}
+
+		$gradient_presets = current( (array) get_theme_support( 'editor-gradient-presets' ) );
+		if ( false !== $gradient_presets ) {
+			$editor_settings['gradients'] = $gradient_presets;
+		}
 
 		/*
 		 * We want the presets and settings declared in theme.json
 		 * to override the ones declared via theme supports.
 		 * So we take theme supports, transform it to theme.json shape
 		 * and merge the static::$theme upon that.
 		 */
-		$theme_support_data = WP_Theme_JSON_Gutenberg::get_from_editor_settings( get_default_block_editor_settings() );
+		$theme_support_data = WP_Theme_JSON_Gutenberg::get_from_editor_settings( $editor_settings );
```

Comment:
What I would suggest is the following:
* Add an optional `$fields` parameter (array of strings) to the `get_default_block_editor_settings()` function.
* Change the function internals to only compute the properties which are requested.
* Pass the new parameter here so that only the above actually needed properties are computed.

Reply:
Broken it up into it's own function with a comment. https://github.com/WordPress/gutenberg/pull/46112/commits/b0a97645995b92f06ea22c413c2b2a1800cc8f3f